test/ecto/changeset/embedded_test.exs

```diff
@@ -463,6 +463,9 @@ defmodule Ecto.Changeset.EmbeddedTest do
     changeset = cast(%Author{}, %{sort: ["new"]}, :posts, opts)
     assert Enum.map(changeset.changes.posts, & &1.changes[:title]) == [nil]
 
+    changeset = cast(%Author{}, %{drop: [0]}, :posts, opts)
+    assert Enum.map(changeset.changes.posts, & &1.changes[:title]) == [nil]
```

Comment:
Isn't this test failing because we are dropping from an empty list and, therefore, there are indeed no changes? What is failing here is the test code, not the actual code?

Can you provide a test code that failed before and passes now? Or is the whole thing an optimization?

Reply:
I opened this PR because I don't see why we check for `is_list(drop)`, i.e. which case it covers.

I'm not a user of these options but I have to mimic these options for `polymorphic_embed` lib, that's how I go into these details:)

Can we find a failing test for the omission of this condition, which would prove that this condition is necessary?

Or maybe you added this option for a matter of consistency (just checking both options), despite that there are probably no cases where a missing key+presence of drop option would succeed.

Reply:
That's a better phrasing of the problem, I was missing the context. I can push a test that shows it working.

Reply:
Thank you for adding the test. Isn't it weird we have empty changes and an action to replace?
The same test with `sort` option instead of `drop` option leads to an error:
```elixir
changeset = cast(%Author{posts: [%Post{title: "one"}]}, %{drop: [9]}, :posts, opts)
IO.inspect changeset.changes

%{posts: [
   #Ecto.Changeset<action: :replace, changes: %{}, errors: [],
    data: #Ecto.Changeset.EmbeddedTest.Post<>, valid?: true, ...>
]}
```

```elixir
changeset = cast(%Author{posts: [%Post{title: "one"}]}, %{sort: ["new"]}, :posts, opts)
IO.inspect changeset.changes

%{}
```

Reply:
I can’t recall the exact conditions but this was 100% added based on scenarios we ran into building LiveView apps.